### ISSUE TITLE
Border-right moving content/not resizing scrollbar-width

### ISSUE THREAD

**Issue by jcdenooy** [KingSora/OverlayScrollbars#95]
First off, thanks heaps for setting up OverlayScrollbars.
Working like a charm, although in some spots I still need to get the hang of it...perhaps a kind of logic :)

I've run test for a superthin scrollbar, also working awesome, except that the people I'm making the site for, have an issue with the (I guess original) scrollbar.

When being rendered it creates a border-right (approx. 30px on a device I did get a chance to test).
I can't reproduce the error on any of my browsers, tried Chrome, Firefox, Opera and Safari – but it doesn't necessarily seem to be a Apple/Windows issue. And it doesn't feel right putting border-right:none !important; on the os-content...

Looking forward to any insight on this. Thanks

**Comment by KingSora** [comment]
Good day!

The `border-right` style gets only added if the browsers scrollbars are already natively overlaid (e.g. they consume 0px of space). 
You can actually test this with `OverlayScrollbars.globals().nativeScrollbarIsOverlaid` ([more info here](https://kingsora.github.io/OverlayScrollbars/#!documentation/gmethod-globals)).

You can actually force Overlaid Scorllbars with you chrome browser, if you type `chrome://flags` in the URL bar, and then enable the `Overlay Scrollbars` feature: [https://i.imgur.com/4PrSPXO.png](https://i.imgur.com/4PrSPXO.png) Maybe you are then able to reproduce it somehow. 

Please check some things for me:
- If you open the demo page with the computer which is having troubles with the plugin, is the same bug happening there on that page too?
- If the `.os-content` element has the `border-right` style applied, does the `.os-viewport` element has a `right: -30px` style applied?
- Does this bug happen only on a certain computer or only on a certain browser on this computer?
- What does the `OverlayScrollbars.globals().nativeScrollbarStyling` property return on the affected browser?

Is it possible for you to send me a screenshot / example with your layout? Maybe I'm able to reproduce it and figure out why this is happening. Please also make sure to not modify the CSS / Style of ANY elements which are added by the plugin except of the `.os-scrollbar` (+ children) and `.os-scrollbar-corner` (+ children) elements.

**Comment by jcdenooy** [comment]
Hi KingSora! Thanks for the quick reply!

> chrome://flags in the URL bar, and then enable the Overlay Scrollbars

On the 'Problem PC' this works. Enabling it solves the issue. Disabling brings it back.
But here I wonder –– that only works on PCs I can access, right? 😦 

> If you open the demo page with the computer which is having troubles with the plugin, is the same bug happening there on that page too?

No, no problems. I tested it with the Nesting Demo-page, as that one is nice and long 😄 Just one question there – the Plugin and Native scrollbars should look different on that page right?

> If the .os-content element has the border-right style applied, does the .os-viewport element has a right: -30px style applied?

Yes

> Does this bug happen only on a certain computer or only on a certain browser on this computer?

Tested on multiple computers – happens on Windows machines, on IE Edge, FireFox, and Chrome

> What does the OverlayScrollbars.globals().nativeScrollbarStyling property return on the affected browser?

I tested _OverlayScrollbars.globals().nativeScrollbarStyling_.y as it is for the Y-axis. Returns FALSE on problem PC. True on my own Mac. `.nativeScrollbarIsOverlaid`returns true on both machines

> What is the display, width and float value of the container to which the plugin is applied?

None set, these are overwritten by the os-host as well, right? I thought because they are overwritten, it's ok not having any set.

> Is the container inside a flexbox or grid layout?

Nope

> Is the direction ltr or rtl?

Left to Right

> Please also make sure to not modify the CSS / Style of ANY elements which are added by the plugin except of the .os-scrollbar

Double-checked, only styled` .os-scrollbar `& kids

> Is it possible for you to send me a screenshot / example with your layout?

Do you want specific elements inspected with this, or certain screens open? OK to send to your email?

**Comment by KingSora** [comment]
Thanks for your response!

> On the 'Problem PC' this works. Enabling it solves the issue. Disabling brings it back.

Yes, this was only for test purposes, it's not a real fix.

> Just one question there – the Plugin and Native scrollbars should look different on that page right?

Yes they should (except if the native scrollbars look like the plugin scrollbars because its simply their native appearance, but thats just a coincidence)

> None set, these are overwritten by the os-host as well, right? I thought because they are overwritten, it's ok not having any set.

Not quite, `.os-host` won't override your styles if you import the Plugin CSS file before your CSS file, but it has it's "basis" styles applied such as `position: relative`.

Please inspect the host element with the DOM unfolded until `.os-content`. You can send everything to my email. If it is possible to you, you can also send the actual website to me, since I've a windows machine too I should be able to reproduce it.

After your first (very detailed) answer, I have now other questions:
- Does the problem machine have a Skin / Theme applied, so scrollbars look across the whole system different than usual?
- If you apply the style `display: block; width: 300px` does it solve the issue? (you can choose any fixed px value you want for width)

I have a theory what the problem could be, but I can't be sure without any real insight...

**Comment by jcdenooy** [comment]
> import the Plugin CSS file before your CSS file

Double-checked, I have the minified css file in front of my main.

> If it is possible to you, you can also send the actual website to me, since I've a windows machine too I should be able to reproduce it.

I'll send it to your email. Thanks a lot 😄 also for the quick response again!!


> Does the problem machine have a Skin / Theme applied, so scrollbars look across the whole system different than usual?

Not that I know of, but I will also include some screenshots – client sent me more strange results from multiple browsers, but I wonder if they're running outdated browsers actually. So for now I prefer to focus on the modern-browser issues.

> If you apply the style display: block; width: 300px does it solve the issue? (you can choose any fixed px value you want for width)
Doesn't seem to solve it, but I'll send you the URL in the email 😉 

> I have a theory what the problem could be, but I can't be sure without any real insight...

Curious to hear, and looking forward to it!! Thanks so much. One email coming your way!

**Comment by KingSora** [comment]
Alright!

After a few e-mail exchanges and insight into the project I've figured out the problem. 
The problem was, that the plugin `JS` and `CSS` wasn't the same version. (js `1.6.3` and css `1.6.0`).

You see, OverlayScrollbars is calculating the native scrollbar size with a generated element (which gets removed instantly after the dimensions are calculated). This elements id is `#os-dummy-scrollbar-size`. Before version `1.6.1` this id was different: `#hs-dummy-scrollbar-size` (a leftover from the time where I haven't decided the name of the plugin). So if you use the incorrect CSS file, this element doesn't have the required styles which are needed to calculate the native scrollbar size correctly. So the calculation ends up always as `natively overlaid` (0px scrollbars) regardless of whether they really are or not. Thus the plugin is applying incorrect styles to certain element, which causes this border issue.

I've also pointed this out (but less detailed) in the release notes for [1.6.1](https://github.com/KingSora/OverlayScrollbars/releases/tag/v1.6.1). 
I'm sorry this kind of unnecessary selector name change has caused you so much trouble, but I really wasn't happy with the old selector as it didn't match up with alle the other selectors.
